src/NzbDrone.Core/Download/Clients/DownloadStation/DownloadStation.cs

```diff
@@ -0,0 +1,323 @@
+                    return DownloadItemStatus.Paused;
+                case DownloadStationTorrent.DownloadStationTaskStatus.Seeding:
+                    return DownloadItemStatus.Completed;
+
+                default:
+                    return DownloadItemStatus.Downloading;
+            }
+        }
+
+        private string GetDefaultDir()
+        {
+            var config = _proxy.GetConfig(Settings);
+
+            var path = config["default_destination"] as string;
+
+            return path;
+        }
+
+        public override DownloadClientStatus GetStatus()
+        {
+            try
+            {
+                var path = GetDownloadDirectory();
+
+                return new DownloadClientStatus
+                {
+                    IsLocalhost = Settings.Host == "127.0.0.1" || Settings.Host == "localhost",
+                    OutputRootFolders = new List<OsPath> { _remotePathMappingService.RemapRemoteToLocal(Settings.Host, new OsPath(path)) }
+                };
+            }
+            catch (DownloadClientException e)
+            {
+                _logger.Debug(e, "Failed to get config from Download Station");
+
+                throw e;
+            } 
+        }
+
+        public override void RemoveItem(string downloadId, bool deleteData)
+        {
+            _proxy.RemoveTorrent(downloadId, deleteData, Settings);            
+        }
+
+        protected override string AddFromMagnetLink(RemoteEpisode remoteEpisode, string hash, string magnetLink)
+        {
+            _proxy.AddTorrentFromUrl(magnetLink, GetDownloadDirectory(), Settings);
+
+            var items = _proxy.GetTorrents(Settings).Where(t => t.Additional.Detail["uri"] == magnetLink);
+
+            if (items.Any())
+            {
+                var item = items.SingleOrDefault();
+
+                if (item != null)
+                {
+                    return item.Id;
+                }
+            }
+            else
+            {
+                _logger.Debug("No such task {0} in Download Station", magnetLink);
+            }
+
+            throw new DownloadClientException("Failed to add task to Download Station");
+        }
+
+        private string GetDownloadDirectory()
+        {
+            if (Settings.TvDirectory.IsNotNullOrWhiteSpace())
+            {
+                return Settings.TvDirectory;
+            }
+            else if (Settings.TvCategory.IsNotNullOrWhiteSpace())
+            {
+                var destDir = GetDefaultDir();
+
+                return $"{destDir.TrimEnd('/')}/{Settings.TvCategory}";
+            }
+
+            return null;
+        }
+
+        protected override string AddFromTorrentFile(RemoteEpisode remoteEpisode, string hash, string filename, byte[] fileContent)
+        {           
+            _proxy.AddTorrentFromData(fileContent, GetDownloadDirectory(), Settings);
+
+            var items = _proxy.GetTorrents(Settings).Where(t => t.Additional.Detail["uri"] == filename);
```

Comment:
I tested it, and i think it's not going to work either. The api doesn't receive a filename, that's why it's not being passed to the proxy...
I need further investigation on this issue